[Windows] minimize window and then reopen cause the layout changed

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. click minimize button to minimize the window (or by click app icon in task bar) -- platform Windows 10
2. click app icon in task bar to reopen the window
3. cause the layout changed (left dock and bottom dock will be more small then before)

### Environment

Zed: v0.165.0 (Zed Dev 57a45d80ad1e3d2b7c87d68fc4e3499527543d1f)
OS: Windows 10.0.19045
Memory: 15.9 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics 620 || Intel Corporation || Intel driver

### If applicable, add mockups / screenshots to help explain present your vision of the feature


https://github.com/user-attachments/assets/ab503a03-44ed-478a-8e68-6c33cd2b286d

debug size from msg after the window minimize
https://github.com/zed-industries/zed/blob/57a45d80ad1e3d2b7c87d68fc4e3499527543d1f/crates/gpui/src/platform/windows/events.rs#L139


https://github.com/user-attachments/assets/a04198c4-1888-4502-bf79-b89f2bfa3d6a



### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-11-30T20:09:27.6942719+08:00 [INFO] ========== starting zed ==========
2024-11-30T20:09:27.7988796+08:00 [INFO] Use Microsoft YaHei UI as UI font.
2024-11-30T20:09:27.8003404+08:00 [INFO] Opening main db
2024-11-30T20:09:27.8027214+08:00 [INFO] Opening main db
2024-11-30T20:09:27.8068585+08:00 [INFO] Using git binary path: None
2024-11-30T20:09:27.8076492+08:00 [ERROR] SHELL environment variable is not assigned so we can't source login environment variables

Caused by:
    environment variable not found
2024-11-30T20:09:27.8731864+08:00 [ERROR] theme not found: Catppuccin Mocha - No Italics
2024-11-30T20:09:27.8736542+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-11-30T20:09:27.8759008+08:00 [INFO] No prompt template overrides directory found at C:\Users\move\AppData\Local\Zed\prompt_overrides. Using built-in prompts.
2024-11-30T20:09:27.8769051+08:00 [INFO] extensions updated. loading 6, reloading 0, unloading 0
2024-11-30T20:09:27.8912088+08:00 [INFO] Opening main db
2024-11-30T20:09:27.8976594+08:00 [ERROR] 找不到元素。 (0x80070490)
2024-11-30T20:09:27.8981472+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-11-30T20:09:27.9225187+08:00 [INFO] Enabling color space support
2024-11-30T20:09:27.9315489+08:00 [INFO] Adapter: "Intel(R) UHD Graphics 620"
2024-11-30T20:09:27.9321155+08:00 [INFO] No ray tracing extensions are supported
2024-11-30T20:09:27.9411955+08:00 [WARN] Requested size 1x1 is outside of surface capabilities
2024-11-30T20:09:27.9412901+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE | INHERIT
2024-11-30T20:09:27.9413774+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:09:27.9414829+08:00 [WARN] Unable to forbid exclusive full screen
2024-11-30T20:09:27.9861807+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-11-30T20:09:27.9863396+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-11-30T20:09:27.9864808+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-11-30T20:09:27.9894547+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9899745+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9906746+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9911005+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.991641+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9920807+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-11-30T20:09:27.9921696+08:00 [INFO] Skip function Some("to_device_position")
2024-11-30T20:09:27.9922751+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-11-30T20:09:27.9923724+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9925714+08:00 [INFO] Skip function Some("blend_color")
2024-11-30T20:09:27.9939119+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9945002+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9949165+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9957659+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9965986+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9971327+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.997527+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-11-30T20:09:27.9976141+08:00 [INFO] Skip function Some("to_device_position")
2024-11-30T20:09:27.9977031+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-11-30T20:09:27.997787+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:27.9979623+08:00 [INFO] Skip function Some("blend_color")
2024-11-30T20:09:28.0003241+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE | INHERIT
2024-11-30T20:09:28.0004219+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:09:28.0005162+08:00 [WARN] Unable to forbid exclusive full screen
2024-11-30T20:09:28.0159414+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:09:28.0160451+08:00 [WARN] Unable to forbid exclusive full screen
2024-11-30T20:09:28.0415597+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-11-30T20:09:28.0443532+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0447522+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0453411+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0457396+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0462827+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0466245+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-11-30T20:09:28.0467129+08:00 [INFO] Skip function Some("to_device_position")
2024-11-30T20:09:28.0468297+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-11-30T20:09:28.0469132+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0470713+08:00 [INFO] Skip function Some("blend_color")
2024-11-30T20:09:28.0477082+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0481795+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.048537+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0492993+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0500174+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0504752+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0508469+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-11-30T20:09:28.0509334+08:00 [INFO] Skip function Some("to_device_position")
2024-11-30T20:09:28.0510168+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-11-30T20:09:28.0511+08:00 [INFO] Skip function Some("to_tile_position")
2024-11-30T20:09:28.0512577+08:00 [INFO] Skip function Some("blend_color")
2024-11-30T20:09:28.0526998+08:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "Intel(R) UHD Graphics 620", driver_name: "Intel Corporation", driver_info: "Intel driver" }
2024-11-30T20:09:28.0544638+08:00 [ERROR] theme not found: Catppuccin Mocha - No Italics
2024-11-30T20:09:28.056547+08:00 [ERROR] theme not found: Catppuccin Mocha - No Italics
2024-11-30T20:09:28.0585706+08:00 [ERROR] theme not found: Catppuccin Mocha - No Italics
2024-11-30T20:09:28.0608775+08:00 [ERROR] theme not found: Catppuccin Mocha - No Italics
2024-11-30T20:09:28.0645619+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:09:28.0646704+08:00 [WARN] Unable to forbid exclusive full screen
2024-11-30T20:09:28.123499+08:00 [INFO] Creating texture 0x2b700657a88 of size 1024x1024x1 and format R8Unorm, name 'atlas', handle 0
2024-11-30T20:09:28.1240735+08:00 [INFO] Creating buffer 0x2b700014b68 of size 65536, name 'chunk-0', handle 1
2024-11-30T20:09:28.1431131+08:00 [INFO] Creating buffer 0x2b77e3f8d38 of size 4096, name 'chunk-0', handle 2
2024-11-30T20:09:28.1432642+08:00 [INFO] Creating buffer 0x2b73b7c5598 of size 6720, name 'chunk-1', handle 3
2024-11-30T20:09:28.1506586+08:00 [INFO] Opening main db
2024-11-30T20:09:28.1567482+08:00 [INFO] Using Windows API for pseudoconsole
2024-11-30T20:09:28.2045064+08:00 [INFO] Creating buffer 0x2b70076c538 of size 4256, name 'chunk-2', handle 4
2024-11-30T20:09:28.2098395+08:00 [INFO] Opening main db
2024-11-30T20:09:28.2318878+08:00 [INFO] Creating buffer 0x2b7009358e8 of size 5040, name 'chunk-3', handle 5
2024-11-30T20:09:29.7045411+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:09:29.7047563+08:00 [WARN] Unable to forbid exclusive full screen
2024-11-30T20:09:29.797696+08:00 [INFO] Creating buffer 0x2b77e5ad538 of size 4096, name 'chunk-4', handle 6
2024-11-30T20:09:29.79782+08:00 [INFO] Creating buffer 0x2b77e5adac8 of size 5040, name 'chunk-5', handle 7
2024-11-30T20:09:29.7979626+08:00 [INFO] Creating buffer 0x2b77e5ac488 of size 4096, name 'chunk-6', handle 8
2024-11-30T20:09:31.1861734+08:00 [INFO] Creating buffer 0x2b77e365338 of size 4096, name 'chunk-7', handle 9
2024-11-30T20:09:31.6402778+08:00 [INFO] Creating buffer 0x2b73bea2118 of size 6048, name 'chunk-8', handle 10
2024-11-30T20:09:32.259073+08:00 [INFO] Creating buffer 0x2b7004e7dc8 of size 28336, name 'chunk-9', handle 11
2024-11-30T20:09:32.2637951+08:00 [INFO] Creating buffer 0x2b73bea26a8 of size 28336, name 'chunk-10', handle 12
2024-11-30T20:09:41.4088205+08:00 [INFO] Creating buffer 0x2b700e97508 of size 23184, name 'chunk-11', handle 13
2024-11-30T20:09:46.063038+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-11-30T20:09:46.0813352+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-11-30T20:09:46.0918076+08:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-11-30T20:09:46.1088155+08:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-11-30T20:09:51.0788063+08:00 [INFO] Creating buffer 0x2b700e97a98 of size 25760, name 'chunk-12', handle 14
2024-11-30T20:10:07.9741811+08:00 [INFO] Creating buffer 0x2b73bea15f8 of size 65536, name 'chunk-1', handle 15
2024-11-30T20:10:07.9781179+08:00 [INFO] Creating buffer 0x2b73bea1068 of size 32928, name 'chunk-13', handle 16
2024-11-30T20:10:07.9786766+08:00 [INFO] Creating buffer 0x2b700c09898 of size 32928, name 'chunk-14', handle 17
2024-11-30T20:10:17.7452878+08:00 [INFO] Creating buffer 0x2b700c0aed8 of size 44576, name 'chunk-15', handle 18
2024-11-30T20:10:17.752405+08:00 [INFO] Creating buffer 0x2b7001f4398 of size 44576, name 'chunk-16', handle 19
2024-11-30T20:10:29.14145+08:00 [INFO] Using surface present mode IMMEDIATE
2024-11-30T20:10:29.141569+08:00 [WARN] Unable to forbid exclusive full screen
```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/crates/gpui/src/platform/windows/events.rs
+++ b/crates/gpui/src/platform/windows/events.rs
@@ -33,7 +33,7 @@ pub(crate) fn handle_msg(
         WM_ACTIVATE => handle_activate_msg(handle, wparam, state_ptr),
         WM_CREATE => handle_create_msg(handle, state_ptr),
         WM_MOVE => handle_move_msg(handle, lparam, state_ptr),
-        WM_SIZE => handle_size_msg(lparam, state_ptr),
+        WM_SIZE => handle_size_msg(wparam, lparam, state_ptr),
         WM_ENTERSIZEMOVE | WM_ENTERMENULOOP => handle_size_move_loop(handle),
         WM_EXITSIZEMOVE | WM_EXITMENULOOP => handle_size_move_loop_exit(handle),
         WM_TIMER => handle_timer_msg(handle, wparam, state_ptr),
@@ -136,7 +136,15 @@ fn handle_move_msg(
     Some(0)
 }
 
-fn handle_size_msg(lparam: LPARAM, state_ptr: Rc<WindowsWindowStatePtr>) -> Option<isize> {
+fn handle_size_msg(
+    wparam: WPARAM,
+    lparam: LPARAM,
+    state_ptr: Rc<WindowsWindowStatePtr>,
+) -> Option<isize> {
+    if wparam.0 == SIZE_MINIMIZED as usize {
+        return Some(0);
+    }
+
     let width = lparam.loword().max(1) as i32;
     let height = lparam.hiword().max(1) as i32;
     let mut lock = state_ptr.state.borrow_mut();

